beacon-chain/blockchain/process_block.go

```diff
@@ -250,51 +238,6 @@ func (s *Service) onBlockBatch(ctx context.Context, blks []*ethpb.SignedBeaconBl
 	return preState, fCheckpoints, jCheckpoints, nil
 }
 
-// handles the state post transition and saves the appropriate checkpoints and forkchoice
-// data.
-func (s *Service) handlePostStateInSync(ctx context.Context, signed *ethpb.SignedBeaconBlock,
-	blockRoot [32]byte, postState *stateTrie.BeaconState) error {
-
-	b := signed.Block
-
-	s.saveInitSyncBlock(blockRoot, signed)
-
-	if err := s.stateGen.SaveState(ctx, blockRoot, postState); err != nil {
-		return errors.Wrap(err, "could not save state")
-	}
-
-	if err := s.insertBlockAndAttestationsToForkChoiceStore(ctx, b, blockRoot, postState); err != nil {
-		return errors.Wrapf(err, "could not insert block %d to fork choice store", b.Slot)
-	}
-
-	// Rate limit how many blocks (2 epochs worth of blocks) a node keeps in the memory.
-	if uint64(len(s.getInitSyncBlocks())) > initialSyncBlockCacheSize {
-		if err := s.beaconDB.SaveBlocks(ctx, s.getInitSyncBlocks()); err != nil {
-			return err
-		}
-		s.clearInitSyncBlocks()
-	}
-
-	// Update finalized check point. Prune the block cache and helper caches on every new finalized epoch.
-	if postState.FinalizedCheckpointEpoch() > s.finalizedCheckpt.Epoch {
-		if err := s.beaconDB.SaveBlocks(ctx, s.getInitSyncBlocks()); err != nil {
```

Comment:
dont we need this in the event of a finalized checkpoint being saved ?

Reply:
We do this above. Saving blocks to DB every 2 epochs. It's not required to have these blocks in DB upon finalized check point. Tested run time as well.

```go
// Rate limit how many blocks (2 epochs worth of blocks) a node keeps in the memory.
	if uint64(len(s.getInitSyncBlocks())) > initialSyncBlockCacheSize {
		if err := s.beaconDB.SaveBlocks(ctx, s.getInitSyncBlocks()); err != nil {
			return err
		}
		s.clearInitSyncBlocks()
	}
```